ECIPs/ECIP-1027.md

```diff
@@ -0,0 +1,218 @@
+which has a are limited execution model and it requires enhancements to start using it. Ethereum turing-complete 
+VM is much more flexible and allows to use this approach for scalability even without running a Hard Fork.
+
+# Specification
+
+## Overview
+
+A Peer configures Geth to follow a separate chain (Sidechain) and specifying a more secure chain (Parent Chain) 
+to make checkpoints through a special smart contract. For simplicity it could be two separate instances of Geth, 
+communicating through RPC API. For better usability it would be optimal to add ability to follow several chains from 
+same Geth instance.
+
+## Key points:
+
+ - There should be a special Smart Contract deployed to Parent Chain to manage checkpoints
+ - Checkpoints are transactions in Parent Chain
+ - Sidechain can read from Parent Chain and verify that it follows longest or most difficult Sidechain
+ - Mainnet is going to work as usual. Not a fork.
+
+It maybe helpful to introduce additional Opcodes to improve cross-transaction functionality. That will require a 
+protocol upgrade, but that’s optional and will be discussed in separate ECIPs
+
+## Types of chains
+
+There can be selected major 3 types of Sidechains:
+
+ - _Private chain_ - uses mainnet only for consensus
+ - _Trusted public chain_ - operated by a trusted party
+ - _Public chain_ - general use blockchain operated in untrusted environment
+
+### Private Chain (C1)
+
+Private Chain is a fully private chain executed in isolated environment which is using Parent Chain only as a security 
+mechanism.
+ 
+This can be a solution for many cases of a private corporate blockchains. This types of chain doesn’t share state with 
+parent chain, and therefore doesn’t need cross transactions, or can perform them in individual way when needed.
+
+Conception of making checkpoints on public secure blockchain is a popular idea in private chains and standalone 
+services, also called as Anchoring.
+
+### Trusted Public Chain (C2)
+
+Trusted Public Chain is a chain managed by a public service or Dapp, and such service provides full management of 
+that chain and performs cross operations for their users. Sidechain can use PoA or PoS for internal consensus, and a 
+separate Oracle/Validator like service to send/confirm user actions performed across the chains. For most of the cases 
+this chain could have a simple shared state, such as buying/selling internal tokens through a parent coin.
+ 
+Smart contracts for controlling shared state, and Transaction Replay like logic can give users additional security for 
+their cross-chain operations.
+
+_Example_: a smart contract based Token Exchange with Deposit/Withdraw through a special Smart Contract in Parent chain
+
+### Public Chain (C3)
+
+Public Chain is a general use blockchain, operated by multiple parties and in an untrusted environment. It can use a 
+fast PoS consensus algorithm for fast and cheap transactions, or individual configuration of EVM.
+
```

Comment:
Is `C1/C2/C3` some common sidechain classification or was invented by you @splix?